lib/job.js

```diff
@@ -1,106 +1,249 @@
 'use strict';
 
+const bluebird = require('bluebird');
+const Path = require('path');
+const logger = require('./logger.js');
+
+function Job(client, config) {
+  this.client = client;
+  this.config = config;
+
+  // Shared between all Job classes
+  this.jid = config.jid;
+  this.priority = config.priority;
+  this.klassName = config.klass;
+  this.queueName = config.queue;
+  this.tags = config.tags;
+  this.data = JSON.parse(config.data);
+
+  // Specific to non-recurring jobs
+  this.state = config.state;
+  this.tracked = config.tracked;
+  this.failure = config.failure;
+  this.history = config.history;
+  this.dependents = config.dependents;
+  this.dependencies = config.dependencies;
+  this.expiresAt = config.expires;
+  this.originalRetries = config.retries;
+  this.retriesLeft = config.remaining;
+  this.workerName = config.worker;
+}
+
 /**
- * The Job class encapsulates the properties of a qless job being worked
- * on by a particular worker (our worker) and allows us to cancel the job,
- * fail the job, etc.
+ * Given a string name of a class, return the corresponding class
+ *
+ * Split the provided name by `/`, and import the longest working prefix. The
+ * remaining components will be treated as attributes off of the import.
+ *
+ * For example, consider 'some/package/file.js/property/class'. We would
```

Comment:
This seems a little hokey to me. Is there a reason to not just look for the `*.js` extension? I don't feel strongly about it, and it appears to work, so maybe it's not a big deal, but it does seem a little complex.